src/codeflare_sdk/ray/rayjobs/rayjob.py

```diff
@@ -282,3 +299,265 @@ def _map_to_codeflare_status(
+            config_builder: ManagedClusterConfig instance for building specs
+        """
+
+        # Get existing RayCluster
+        api_instance = client.CustomObjectsApi(get_api_client())
+        try:
+            ray_cluster = api_instance.get_namespaced_custom_object(
+                group="ray.io",
+                version="v1",
+                namespace=self.namespace,
+                plural="rayclusters",
+                name=self.cluster_name,
+            )
+        except client.ApiException as e:
+            raise RuntimeError(f"Failed to get RayCluster '{self.cluster_name}': {e}")
+
+        # Build script volume and mount specifications using config.py
+        script_volume, script_mount = config_builder.build_script_volume_specs(
+            configmap_name=configmap_name, mount_path="/home/ray/scripts"
+        )
+
+        # Helper function to check for duplicate volumes/mounts
+        def volume_exists(volumes_list, volume_name):
+            return any(v.get("name") == volume_name for v in volumes_list)
+
+        def mount_exists(mounts_list, mount_name):
+            return any(m.get("name") == mount_name for m in mounts_list)
+
+        # Add volumes and mounts to head group
+        head_spec = ray_cluster["spec"]["headGroupSpec"]["template"]["spec"]
+        if "volumes" not in head_spec:
+            head_spec["volumes"] = []
+        if not volume_exists(head_spec["volumes"], script_volume["name"]):
+            head_spec["volumes"].append(script_volume)
+
+        head_container = head_spec["containers"][0]  # Ray head container
+        if "volumeMounts" not in head_container:
+            head_container["volumeMounts"] = []
+        if not mount_exists(head_container["volumeMounts"], script_mount["name"]):
+            head_container["volumeMounts"].append(script_mount)
+
+        # Add volumes and mounts to worker groups
+        for worker_group in ray_cluster["spec"]["workerGroupSpecs"]:
+            worker_spec = worker_group["template"]["spec"]
+            if "volumes" not in worker_spec:
+                worker_spec["volumes"] = []
+            if not volume_exists(worker_spec["volumes"], script_volume["name"]):
+                worker_spec["volumes"].append(script_volume)
+
+            worker_container = worker_spec["containers"][0]  # Ray worker container
+            if "volumeMounts" not in worker_container:
+                worker_container["volumeMounts"] = []
+            if not mount_exists(worker_container["volumeMounts"], script_mount["name"]):
+                worker_container["volumeMounts"].append(script_mount)
+
+        # Update the RayCluster
+        try:
+            api_instance.patch_namespaced_custom_object(
+                group="ray.io",
+                version="v1",
+                namespace=self.namespace,
+                plural="rayclusters",
+                name=self.cluster_name,
+                body=ray_cluster,
+            )
```

Comment:
```suggestion
            self._cluster_api_.patch_ray_cluster(
                name=self.cluster_name,
                ray_patch: ??? # unsure here as the kuberay client code here is funny, i will investigate and get back to you
                k8s_namespace=self.namespace,
            )
```

Reply:
on the above ^ this will work, just pass in the ray cluster object into `ray_patch` like you're passing it into `body`